index.js

```diff
@@ -26,15 +30,27 @@ ACOSAPlus.addToBody = function(params, req) {
   if (req.query.content !== 'ready') {
     var hostUrl = req.protocol + '://' + req.get('host');
     var fullUrl = hostUrl + req.originalUrl + '&content=ready';
-    var width = req.query.width || 770;
+    var width = req.query.width || 600;
     var height = req.query.height || 500;
-    params.bodyContent += '<iframe class="acos-iframe" src="' + fullUrl + '" width="' + width + '" height="' + height + '" style="box-shadow: none; border: none;"></iframe>\n';
-    if (!req.query.noResizeIframe) {
+    if (req.query.noResizeIframe) {
+      params.bodyContent += '<iframe class="acos-iframe" src="' + fullUrl + '" width="' + width + '" height="' + height + '" style="box-shadow: none; border: none;"></iframe>\n';
+    } else {
+      params.bodyContent += '<iframe class="acos-iframe" src="' + fullUrl + '" style="box-shadow: none; border: none; width: 100%;"></iframe>\n';
       params.bodyContent += '<script src="' + hostUrl + '/static/aplus/resizeiframe.js" type="text/javascript"></script>\n';
     }
   } else {
     // This will be inside the previously created iframe
     params.bodyContent += '<input type="hidden" name="submission_url" value="' + req.query.submission_url + '">\n';
+    
+    // Hash the user ID (uid) parameter so that, if it is written to ACOS logs,
+    // it is not possible to trace it back to the A+ user id and the real identity of the user.
+    // However, the same user has the same ID in the ACOS logs across exercises so that
+    // the user's progress in the course may be investigated.
+    var uid = req.query.uid;
+    if (uid) {
+      var hash = crypto.createHash('sha1').update(uid);
```

Comment:
This may not be the wanted behavior in all cases. If there are some research setups, it can be vital that you can combine logs and other data from multiple sources and then there must be a common unique key to map the data and the real student.

Reply:
I was told to protect the user's privacy too so that's why the logs do not contain the directly recognizable user IDs. The code could be changed for a specific research setup to track the real UID. However, should it use the real UID by default in all cases?

Reply:
I'd suggest to keep the `uid` unhashed because it is not a student number. You need access to A+ in order to map that integer to a specific student. However, if this information is needed, it is available.

Reply:
To protect privacy, it is always possible to setup a private ACOS server or prevent logging by using that parameter.

Reply:
OK, I will change it to the unhashed `uid`.